CNV-39983: Add "virtctl create vm" to the instructions

jhradilek opened the pull request:
<!--- PR title format: [GH#<gh-issue-id>][BZ#<bz-issue-id>][OCPBUGS#<jira-issue-id>][OSDOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

<!--- If your changes apply to the latest release and/or in-development version of OpenShift, open your PR against the `main` branch.
Do not create or rename a top-level directory (or any subdirectory in a directory that contains a hugebook.flag file) in the repository and topic map without checking with a docs program manager first.
If a book is being created or modified, there are changes on the Customer Portal that must also be made.

* For more details about the information requested in this template, see:
  https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/create_or_edit_content.adoc#submit-PR --->

Version(s): 4.18+
<!--- Specify the version or versions of OpenShift your PR applies to. -->

Issue: [CNV-39983](https://issues.redhat.com//browse/CNV-39983)
<!--- Add a link to the Bugzilla, Jira, or GitHub issue, if applicable. --->

Link to docs preview:
- https://89618--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/creating_vms_advanced/creating_vms_cli/virt-creating-vms-from-cli.html#virt-creating-vm-cli_virt-creating-vms-cli
- https://89618--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/creating_vms_advanced/creating_vms_advanced_web/virt-creating-vms-from-web-images.html#virt-creating-vm-import-cli_virt-creating-vms-from-web-images
- https://89618--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/creating_vms_advanced/creating_vms_cli/virt-creating-vms-from-container-disks.html#virt-creating-vm-import-cli_virt-creating-vms-from-container-disks
- https://89618--ocpdocs-pr.netlify.app/openshift-enterprise/latest/virt/creating_vms_advanced/creating_vms_cli/virt-creating-vms-by-cloning-pvcs.html#virt-creating-vm-cloning-pvc-data-volume-template_virt-creating-vms-by-cloning-pvcs
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

QE review:
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

<!--- After you open your PR, ask for review from the OpenShift docs team:
  For community authors: Tag @openshift/team-documentation in a GitHub comment.--->

jhradilek committed:
CNV-39983: Added "virtctl create vm" to the instructions

jhradilek commented:
Note: Some of the content that I am editing has been restructured for 4.18. To backport the changes all the way to 4.14, I will need to create a separate pull request.

jhradilek commented:
Note: The contents for 4.18 onwards is drastically different from the content in 4.14-4.17. I am therefore restricting this pull-request to versions that can be cherry-picked and will create a separate one for the older version.

jhradilek commented:
/label peer-review-needed

I am not sure if the manual merges need to be peer reviewed separately:

https://github.com/openshift/openshift-docs/pull/91053 (4.17)
https://github.com/openshift/openshift-docs/pull/91073 (4.15, 4.16)

bergerhoffer commented:
/label peer-review-in-progress

bergerhoffer commented on modules/virt-creating-vm-cloned-pvc-data-volume-template.adoc:
```diff
@@ -1,73 +1,75 @@
 .Prerequisites
 
 * The VM with the source PVC must be powered down.
+* You must have the `virtctl` command line tool installed.
```
```suggestion
* You must have the `virtctl` command-line tool installed.
```

jhradilek replied:
Thank you. For the record, "comman-line" is also my preference, but in the openshift-docs repository, it seems to be a coin toss and what ultimately convinced me to leave a space was how it is introduced in the OpenShift web console.

```
$ grep -hroe 'command[ -]line tool' | sort | uniq -c
     22 command line tool
     23 command-line tool
```

bergerhoffer replied:
That is a bummer that there are so many incorrect instances in the repo :disappointed: 

Per ISG it definitely should be hyphenated. I'll make a note of it and if I ever get a chance, I'll do a little sweep through to fix them all. Thanks for letting me know!

jhradilek replied:
I can help with the sweep through. I'll open an issue for the OpenShift web console where it is listed as "Command Line Tools" both in the menu and in the section title.

bergerhoffer replied:
Wonderful, ty!

I actually just did a quick sweep and opened this PR: https://github.com/openshift/openshift-docs/pull/91919.

I fixed instances of where "tool" or "interface" comes after it. "command line" on it own would be fine.

I still need to go through and check to make sure they're all okay, but it's a start at least. I can tag you in the PR after I'm done if you don't mind reviewing as well?

jhradilek replied:
Of course, I'll be very happy to review it, thank you.

bergerhoffer commented on modules/virt-creating-vm-cli.adoc:
```diff
@@ -1,16 +1,27 @@
+
+.Prerequisites
+
+* You must have the `virtctl` command line tool installed.
```
```suggestion
* You must have the `virtctl` command-line tool installed.
```

bergerhoffer commented on modules/virt-creating-vm-cloned-pvc-data-volume-template.adoc:
```diff
@@ -1,73 +1,75 @@
 [source,terminal]
 ----
-$ oc create -f <vm-clone-datavolumetemplate>.yaml
+$ oc create -f <vm_manifest_file>.yaml
```
I know it was already like this, but I was wondering whether the first step actually saves the manifest as a file already for you, or if it just outputs the YAML to the screen and then you have to copy/paste/save it as a file? 

My reason for asking is 1). it could help to clarify this / maybe it's missing a step, and 2). could we actually show the above YAML example as a particular example file name, and then just use that here?

(Same comment for the other modules, if anything does happen to get updated)

bergerhoffer commented on modules/virt-creating-vm-cloned-pvc-data-volume-template.adoc:
```diff
@@ -1,73 +1,75 @@
+<1> The VM name.
+<2> The name of the source PVC.
+<3> The namespace of the source PVC.
+<4> If the PVC source has appropriate labels, the instance type can be inferred from the selected `DataSource` object.
```
Could "can be" be changed to "is" for this and the next line? Or would that not be actually accurate?

bergerhoffer commented on modules/virt-creating-vm-container-disk-cli.adoc:
```diff
@@ -0,0 +1,81 @@
+.Prerequisites
+
+* You must have access credentials for the container registry that contains the container disk.
+* You must have the `virtctl` command line tool installed.
```
```suggestion
* You must have the `virtctl` command-line tool installed.
```

bergerhoffer commented on modules/virt-creating-vm-container-disk-cli.adoc:
```diff
@@ -0,0 +1,81 @@
+
+. Review the `VirtualMachine` manifest for your VM:
++
+[%collapsible]
```
This added a collapsible section, but the other modules don't have this. Should it be consistent one way or the other across them?

jhradilek replied:
That's a very good point, the collapsible attribute is inherited from the module I split. but considering the updated manifest is fairly straightforward and fits on one screen, I don't think it is needed anywhere.

bergerhoffer commented on modules/virt-creating-vm-container-disk-cli.adoc:
```diff
@@ -0,0 +1,81 @@
+
+.Verification
+
+. Monitor the VM until its status is `Running` by running the following command:
```
Reading "`Running` by running" sounds repetitive. Could this be reworded to not use back-to-back "running"?

Could either reword, or you could add some example output after the command, or just some text that says to verify that the text is `Running`. (Not sure if it's in a certain column, or whether that's the only value shown from this command)

bergerhoffer commented on modules/virt-creating-vm-container-disk-cli.adoc:
```diff
@@ -0,0 +1,81 @@
+$ oc get vm <vm_name>
+----
+
+. Verify that provisioning is complete and that the VM has started by accessing its serial console:
```
This could probably also use the "by running the following command" for consistency with the above step (though you might have to reword a little since there is already a "by").

Maybe it'll be obvious when they run the command, but perhaps some example output or text afterward telling users what to look for to know if it's complete could be helpful. (Or does this launch a UI console or something where they check for it??)

bergerhoffer commented on modules/virt-creating-vm-web-page-cli.adoc:
```diff
@@ -0,0 +1,105 @@
+.Prerequisites
+
+* You must have access credentials for the web page that contains the image.
+* You must have the `virtctl` command line tool installed.
```
```suggestion
* You must have the `virtctl` command-line tool installed.
```

bergerhoffer commented on modules/virt-creating-vm-web-page-cli.adoc:
```diff
@@ -0,0 +1,105 @@
+// * virt/creating_vms_advanced/creating_vms_advanced_web/virt-creating-vms-from-web-images.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="virt-creating-vm-import-cli_{context}"]
```
This ID should probably be updated to match the file name

jhradilek replied:
I initially did but this module is referenced multiple times and possibly linked form external sources. This was the least invasive way, but if you insist, I can change it. The cherry-picks will be a nightmare.

bergerhoffer replied:
I don't insist :)

jhradilek replied:
Thank you! And thank you very much for your thorough review, I am working on fixing the other parts and will retest the commands tomorrow to get expected output for verification steps.

bergerhoffer commented on modules/virt-creating-vm-web-page-cli.adoc:
```diff
@@ -0,0 +1,105 @@
+
+You can create a virtual machine (VM) from an image on a web page by using the command line.
+
+When the virtual machine (VM) is created, the data volume with the image is imported into persistent storage.
```
Since you added it in parentheses above, I don't think you need to expand it again

```suggestion
When the VM is created, the data volume with the image is imported into persistent storage.
```

bergerhoffer commented on modules/virt-creating-vm-web-page-cli.adoc:
```diff
@@ -0,0 +1,105 @@
+spec:
+  dataVolumeTemplates:
+  - metadata:
+      name: imported-volume-6dcpf # <5>
```
This threw me off that it went from callout 1 to 5. I think normally we'd want to see them go in numerical order. So I'd recommend updating the callout numbers here and below to go in order.

jhradilek replied:
That is a relic from when the module overused conditions, I will re-order the callouts.

bergerhoffer commented on modules/virt-creating-vm-web-page-cli.adoc:
```diff
@@ -0,0 +1,105 @@
+$ oc describe dv <data_volume_name>
+----
+
+. Verify that provisioning is complete and that the VM has started by accessing its serial console:
```
Same comment as the other module about what you expect to see

bergerhoffer reviewed (commented):
A few thoughts/suggestions!

jhradilek commented:
Hi @bergerhoffer I apologize for the delay. I addressed most of your comments withe the exception of the id and the placeholders, which I addressed in the comments above. This actually took longer than expected because your great comments made me re-test the verification steps and subsequently, we changed one of the commands with the SME. Would you mind taking a quick look if my changes are in line with what you had in mind? If so, I am going to apply them on the pull requests for 4.15-4.17 as well.

bergerhoffer commented:
> Hi @bergerhoffer I apologize for the delay. I addressed most of your comments withe the exception of the id and the placeholders, which I addressed in the comments above. This actually took longer than expected because your great comments made me re-test the verification steps and subsequently, we changed one of the commands with the SME. Would you mind taking a quick look if my changes are in line with what you had in mind? If so, I am going to apply them on the pull requests for 4.15-4.17 as well.

@jhradilek That's great to hear! Your updates all lgtm :)

jhradilek commented:
/label merge-review-needed

There are two manual merges associated with this pull request as the files have diverged between versions and automatic cherry-picks wouldn't work:

https://github.com/openshift/openshift-docs/pull/91053 (4.17)
https://github.com/openshift/openshift-docs/pull/91073 (4.15, 4.16)

I applied changes suggested by Andrea to all of them.

mburke5678 commented on modules/virt-creating-vm-cli.adoc:
```diff
@@ -1,16 +1,27 @@
 .Procedure
 
-. Edit the `VirtualMachine` manifest for your VM. The following example configures a {op-system-base-full} VM:
+. Create a `VirtualMachine` manifest for your VM and save it as a YAML file. For example, to create a minimal {op-system-base-full} VM, run the following command:
```
I find this a bit confusing. In the first sentence, you have me creating a YAML manifest. Then, in the second, you say I can create a VM using the command. Looking up the command on line, I understand what is going on. I don't want to hold up the merge over it; but, maybe for a future change:

 ```suggestion
. Create a `VirtualMachine` manifest for your VM by using the `virtctl crate vm` command and save it as a YAML file. For example, to create a manifest for a minimal {op-system-base-full} VM, run the following command:
```

mburke5678 commented:
/cherrypick enterprise-4.18

mburke5678 commented:
/cherrypick enterprise-4.19
